### DISCUSSION TITLE
How to put a custom ID to the collections with this nodes?

### DISCUSSION THREAD

**Discussion by Alvarum** [GogoVega/node-red-contrib-firebase-realtime-database#27] [Q&A]
Hi everyone, im trying to put some custom ID to the collections i send to Firebase, with this node i din't find at the moment how to do it, as an example of what im trying to say: 

`var docRef = db.collection('objects').doc('some-id');`

In that code, given in the documntation of Firebase, the `.doc('some-id')` is the part that adds de ID. By default this node uses the automatic id generation of Firebase that put this kind of id's `-NR8dN0yIIhikzwFapup`.

This is the code i got to create one:
```
let array = msg.payload.split(','),
    placa = array[0], macrca = array[1], color = array[2], hora = array[3], fecha = array[4]
let patente ={ 
        "Placa": placa,
        "Marca": macrca,
        "Color": color,
        "Hora": hora}        
msg.payload = patente
msg.topic = fecha;
return msg;
```
and this is how the direction is set for the propety "Color": 

`/2023-03-22/-NR8dN0yIIhikzwFapup/Color`

is there any way to do this with this node?

**Comment by GogoVega** [comment]
Hi,
`collection` is a structure used for the **Firestore** database because there are no tables or rows (NoSQL). The real-time database has a JSON structure.

You can structure your data with the real-time database directly from the `payload` you send.
You can learn more by reading this [guide](https://firebase.google.com/docs/database/web/structure-data?hl=en).

Then you can use [QueryConstraints](https://firebase.google.com/docs/database/web/lists-of-data?hl=en#sorting_and_filtering_data) to sort and order the data you read.

**Reply by GogoVega** [reply]
Hi @Alvarum, did that answer your question?

**Reply by Alvarum** [reply]
Hi @GogoVegeta, no, but I managed to find the solution, I'll comment it here when I have some free time, but thanks for your answer! the guide help me a lot too

**Reply by GogoVega** [reply]
Hi @Alvarum,
Thanks for your message, could you see if the [wiki](https://github.com/GogoVega/node-red-contrib-firebase-realtime-database/wiki) answers your question, or what should I add? Thanks.
